### PR TITLE
xz-utils security advisory

### PR DESCRIPTION
**Pull request by afayasa**
## Describe the Change

This PR adds the security advisory for recent critical CVE on x-utils package used in our OS images.

## Changed Pages

💻 [Preview URL](https://deploy-preview-2557--docs-spectrocloud.netlify.app/security-bulletins/cve-reports#april-2-2024---cve-2024-3094-malicious-code-was-discovered-in-the-upstream-tarballs-of-xz---10-cvss)

## Jira Tickets

No JIRA ticket.

## Backports

Not required

### CODE REVIEW

File: docs/docs-content/security-bulletins/cve-reports.md [hunk 1]
```diff
@@ -32,6 +32,35 @@ _Are there any links users can visit to find out more?_
...
+
+Malicious code was discovered in the upstream tarballs of xz, starting with version 5.6.0. Through a series of complex 
+obfuscations, the liblzma build process extracts a prebuilt object file from a disguised test file existing in the 
+source code, which is then used to modify specific functions in the liblzma code. 
+This results in a modified liblzma library that can be used by any software linked against this library, intercepting 
+and modifying the data interaction with this library.
+
+#### Impact
+
+No impact. Palette uses Ubuntu 22.04 OS images while VerteX uses Ubuntu 20.04 OS images for Enterprise,
```
Comment (karl-cardenas-coding) [inline docs/docs-content/security-bulletins/cve-reports.md:45]: Does this refer to self-hosted Palette and VerteX? What about non-hosted environments, such as workload clusters?

Is this also true for someone running version 3.4?

File: docs/docs-content/security-bulletins/cve-reports.md [hunk 2]
```diff
@@ -32,6 +32,35 @@ _Are there any links users can visit to find out more?_
 
 -->
 
+## April 2, 2024 - CVE-2024-3094 Malicious code was discovered in the upstream tarballs of xz - 10 CVSS
```
Comment (karl-cardenas-coding) [inline docs/docs-content/security-bulletins/cve-reports.md:35]:
```suggestion
## April 2, 2024 - CVE-2024-3094 Malicious Code in XZ Library - 10 CVSS
```

File: docs/docs-content/security-bulletins/cve-reports.md [hunk 3]
```diff
@@ -32,6 +32,35 @@ _Are there any links users can visit to find out more?_
...
+Malicious code was discovered in the upstream tarballs of xz, starting with version 5.6.0. Through a series of complex 
+obfuscations, the liblzma build process extracts a prebuilt object file from a disguised test file existing in the 
+source code, which is then used to modify specific functions in the liblzma code. 
+This results in a modified liblzma library that can be used by any software linked against this library, intercepting 
+and modifying the data interaction with this library.
+
+#### Impact
+
+No impact. Palette uses Ubuntu 22.04 OS images while VerteX uses Ubuntu 20.04 OS images for Enterprise, 
+Private Cloud Gateway and Tenant clusters. Both Ubuntu 22.04 and Ubuntu 20.04 images uses an older non-impacted
```
Comment (karl-cardenas-coding) [inline docs/docs-content/security-bulletins/cve-reports.md:46]: Can we share what version, just so that's more clear?

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
Questions

File: docs/docs-content/security-bulletins/cve-reports.md (hunk 1 above)
Comment (afayasa) [reply docs/docs-content/security-bulletins/cve-reports.md:45]: Tenant cluster is mentioned in that statement. We can change it to workload cluster if that's how we are referring them.
Yes, it is true for all versions supported.

File: docs/docs-content/security-bulletins/cve-reports.md (hunk 3 above)
Comment (afayasa) [reply docs/docs-content/security-bulletins/cve-reports.md:46]: Installed version may differ depending on the image. For example, OS image from 3.4 or 4.0 will have different version of xz-utils than the ones from 4.2 or 4.3.

File: docs/docs-content/security-bulletins/cve-reports.md [hunk 4]
```diff
@@ -32,6 +32,39 @@ _Are there any links users can visit to find out more?_
...
+This results in a modified liblzma library that can be used by any software linked against this library, intercepting 
+and modifying the data interaction with this library.
+
+#### Impact
+
+No impact. Palette supports below OS distributions and none of them uses the impacted versions of xz-utils package.
+- [Ubuntu 20.04/22.04/23.04](https://ubuntu.com/security/CVE-2024-3094)
+- [RHEL 8](https://www.redhat.com/en/blog/urgent-security-alert-fedora-41-and-rawhide-users)
+- [OpenSUSE Leap](https://www.suse.com/c/suse-addresses-supply-chain-attack-against-xz-compression-library/)
+- [SLES Micro](https://www.suse.com/c/suse-addresses-supply-chain-attack-against-xz-compression-library/)
```
Comment (jzhoucliqr) [inline docs/docs-content/security-bulletins/cve-reports.md:49]:
```suggestion
- [SLE Micro](https://www.suse.com/c/suse-addresses-supply-chain-attack-against-xz-compression-library/)
```

**Review by karl-cardenas-coding — APPROVED** [review]

**Pull request merged**